cpp/oneapi/dal/algo/kmeans/backend/gpu/infer_kernel_dpc.cpp

```diff
@@ -81,16 +82,20 @@ struct infer_kernel_gpu<Float, method::lloyd_dense, task::clustering> {
                                                arr_distance_block,
                                                arr_closest_distances,
                                                { data_squares_event, centroid_squares_event });
-        kernels_fp<Float>::compute_objective_function(queue,
-                                                      arr_closest_distances,
-                                                      arr_objective_function,
-                                                      { assign_event })
-            .wait_and_throw();
-
-        return infer_result<task::clustering>()
-            .set_responses(dal::homogen_table::wrap(arr_responses.flatten(queue), row_count, 1))
-            .set_objective_function_value(
+        if (desc.get_result_options().test(result_options::compute_exact_objective_function)) {
+            kernels_fp<Float>::compute_objective_function(queue,
+                                                          arr_closest_distances,
+                                                          arr_objective_function,
+                                                          { assign_event })
+                .wait_and_throw();
+            result.set_objective_function_value(
                 static_cast<double>(*arr_objective_function.to_host(queue).get_data()));
+        }
+        if (desc.get_result_options().test(result_options::compute_assignments)) {
+            result.set_responses(
```

Comment:
Probably need assign_event wait here or some sort of event handling since its only used as event dependency in conditional statement above and is probably necessary before set_responses

Reply:
thanks for comment! Fixed